app/gui2/src/components/CreateNewNode.vue

```diff
@@ -0,0 +1,104 @@
+<script setup lang="ts">
+import SvgButton from '@/components/SvgButton.vue'
+import { useVisualizationConfig } from '@/providers/visualizationConfig'
+import { Ast } from '@/util/ast'
+import { Pattern } from '@/util/ast/match'
+import type { NodeCreationOptions } from './GraphEditor/nodeCreation'
+
+type SortDirection = 'asc' | 'desc'
+export type SortModel = {
+  columnName: string
+  sortDirection: SortDirection
+  sortIndex: number
+}
+
+const props = defineProps<{
+  filterModel: {
+    [key: string]: {
+      values: any[]
+      filterType: string
+    }
+  }
+  sortModel: SortModel[]
+}>()
+
+const config = useVisualizationConfig()
+
+const makeSortPattern = () => {
+  const sortMapping = {
+    asc: '..Ascending',
+    desc: '..Descending',
+  }
+  return props.sortModel
+    .filter((sort) => sort?.columnName)
+    .sort((a, b) => a.sortIndex - b.sortIndex)
+    .map((sort) => {
+      return `(..Name '${sort.columnName}' ${sortMapping[sort.sortDirection]})`
+    })
+}
+
+function getAstPatternSort() {
+  return Pattern.new((ast) =>
+    Ast.App.positional(
+      Ast.PropertyAccess.new(ast.module, ast, Ast.identifier('sort')!),
+      Ast.parse(`[${makeSortPattern()}]`),
```

Comment:
It'st best to avoid generating code as a string and then parsing it. Constructing the tree node by node can be tedious, but we have a `Pattern.parse` that can help build complex expressions safely. This is similar to using parameterized queries for SQL. In this case it would look something like:
```
// Even though this doesn't have reactive inputs, it is a `computed` so that it will be created lazily.
const sortPatternPattern = computed(() => Pattern.parse('..Name __ __'))
const sortDirection = computed(() =>( {
  asc: Ast.AutoScope.new(Ast.Ident.new('Ascending')),
  desc: Ast.AutoScope.new(Ast.Ident.new('Descending')),
}))
const makeSortPattern = () => {
  const columnSortExpressions = props.sortModel
    .filter((sort) => sort?.columnName)
    .sort((a, b) => a.sortIndex - b.sortIndex)
    .map((sort) => sortPatternPattern.value.instantiateCopied([Ast.TextLiteral.new(sort.columnName), sortDirection.value[sort.sortDirection]])
    )
  return Ast.Vector.new(columnSortExpressions)
}
```
The APIs above are off the top of my head so if anything isn't making sense let me know.